store/sqlstore/terms_of_service_store.go

```diff
@@ -40,107 +40,81 @@ func NewSqlTermsOfServiceStore(sqlStore SqlStore, metrics einterfaces.MetricsInt
 }
 
-func (s SqlTermsOfServiceStore) Save(termsOfService *model.TermsOfService) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if len(termsOfService.Id) > 0 {
-			result.Err = model.NewAppError(
-				"SqlTermsOfServiceStore.Save",
-				"store.sql_terms_of_service_store.save.existing.app_error",
-				nil,
-				"id="+termsOfService.Id, http.StatusBadRequest,
-			)
-			return
-		}
+func (s SqlTermsOfServiceStore) Save(termsOfService *model.TermsOfService) (*model.TermsOfService, *model.AppError) {
+	if len(termsOfService.Id) > 0 {
+		return nil, model.NewAppError("SqlTermsOfServiceStore.Save", "store.sql_terms_of_service_store.save.existing.app_error", nil, "id="+termsOfService.Id, http.StatusBadRequest)
+	}
 
-		termsOfService.PreSave()
+	termsOfService.PreSave()
 
-		if result.Err = termsOfService.IsValid(); result.Err != nil {
-			return
-		}
+	if err := termsOfService.IsValid(); err != nil {
+		return nil, err
+	}
 
-		if err := s.GetMaster().Insert(termsOfService); err != nil {
-			result.Err = model.NewAppError(
-				"SqlTermsOfServiceStore.Save",
-				"store.sql_terms_of_service.save.app_error",
-				nil,
-				"terms_of_service_id="+termsOfService.Id+",err="+err.Error(),
-				http.StatusInternalServerError,
-			)
-		}
+	if err := s.GetMaster().Insert(termsOfService); err != nil {
+		return nil, model.NewAppError("SqlTermsOfServiceStore.Save", "store.sql_terms_of_service.save.app_error", nil, "terms_of_service_id="+termsOfService.Id+",err="+err.Error(), http.StatusInternalServerError)
+	}
 
-		result.Data = termsOfService
+	termsOfServiceCache.AddWithDefaultExpires(termsOfService.Id, termsOfService)
 
-		termsOfServiceCache.AddWithDefaultExpires(termsOfService.Id, termsOfService)
-	})
+	return termsOfService, nil
 }
 
-func (s SqlTermsOfServiceStore) GetLatest(allowFromCache bool) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if allowFromCache {
-			if termsOfServiceCache.Len() == 0 {
+func (s SqlTermsOfServiceStore) GetLatest(allowFromCache bool) (*model.TermsOfService, *model.AppError) {
+	if allowFromCache {
+		if termsOfServiceCache.Len() != 0 {
+			if cacheItem, ok := termsOfServiceCache.Get(termsOfServiceCache.Keys()[0]); ok {
 				if s.metrics != nil {
-					s.metrics.IncrementMemCacheMissCounter(termsOfServiceCacheName)
-				}
-			} else {
-				if cacheItem, ok := termsOfServiceCache.Get(termsOfServiceCache.Keys()[0]); ok {
-					if s.metrics != nil {
-						s.metrics.IncrementMemCacheHitCounter(termsOfServiceCacheName)
-					}
-
-					result.Data = cacheItem.(*model.TermsOfService)
-					return
-				} else if s.metrics != nil {
-					s.metrics.IncrementMemCacheMissCounter(termsOfServiceCacheName)
+					s.metrics.IncrementMemCacheHitCounter(termsOfServiceCacheName)
 				}
+
+				return cacheItem.(*model.TermsOfService), nil
 			}
 		}
+	}
 
-		var termsOfService *model.TermsOfService
+	if s.metrics != nil {
+		s.metrics.IncrementMemCacheMissCounter(termsOfServiceCacheName)
```

Comment:
This appears to be a change in behaviour: the miss counter will be incremented even when `allowFromCache` is `false`, but I don't think that should be counted as such.

Reply:
Yes, you are right, I was trying to add consistency to this because in other places we count that as a cache miss. I think this should be the correct behavior. @jwilander probably you are more familiarized with the original decision about it. A "allow_cache = false" trigger a cache miss count?

Reply:
@crspeller would actually be the best person to ask. Generally I think it should only be a cache miss if we're trying to use the cache but I could see it being useful to know how often we're not using the cache as well. That might need some re-thinking but I'd push for consistency with what we're doing in other stores right now

Reply:
I think I agree with @jespino here. I miss should be any time we don't use the cache because that's the interesting metric. We want to see how often the cache is used overall. Not when we tried and failed. Although it might be a good improvement to capture all three cases.